python/pyspark/ml/param/__init__.py

```diff
@@ -147,10 +147,10 @@ def hasParam(self, paramName):
     def getOrDefault(self, param):
         """
         Gets the value of a param in the user-supplied param map or its
-        default value. Raises an error if either is set.
+        default value. Raises an error if neither is set.
         """
         param = self._resolveParam(param)
-        if param in self._paramMap:
+        if param in self._paramMap and self.paramMap[param]:
```

Comment:
I don't think the value could be None unless the user explicitly sets it to be None, in which case we should probably assume that "None" is a valid value.